src/shared/components/common/user-badges.tsx

```diff
@@ -43,13 +42,24 @@ function getRoleLabelPill({
 @tippyMixin
 export class UserBadges extends Component<UserBadgesProps> {
   render() {
+    const isBot = this.props.person?.bot_account;
+    const isDeleted = this.props.person?.deleted;
+    var isNewAccount = false;
+    if (this.props.person !== undefined) {
+      // 7 days ago
+      // https://stackoverflow.com/a/563442
+      var date = new Date();
+      date.setDate(date.getDate() - 7);
+      isNewAccount = new Date(this.props.person?.published_at) > date;
+    }
```

Comment:
Externalize this to a function called `isNewAccount(`, and do something like.

```typescript
const isNewAccount = person ? isNewAccount(this.props.person.published_at) : false;
```

Also use `let` instead of `var` in typescript.

Reply:
Done. Made the method a bit more general to use in another PR as well.